src/components/annotations/attributes.js

```diff
@@ -21,6 +20,7 @@ module.exports = {
         valType: 'boolean',
         role: 'info',
         dflt: true,
+        editType: 'docalcAutorange',
```

Comment:
Would `docalc+autorange` be a sensible name for this? It sounds like an enum camelcased and concatenated, but I could be mistaken.

Reply:
Ah, I see. Unless I'm mistaken, this adds no new labels. That's just plugging into what already exists. Still seems like maybe an enum in disguise, but nothing to change here then.

Reply:
One additional note while digging through here is that the `do-` prefix seems to be carried forward from what was originally just internal variable names. Would `editType: 'calcAutorange'` or `editType: 'calc+autorange'` make sense? It seems a bit friendlier if you're trying to make sense of the schema as a standalone concept.

Reply:
`docalcAutorange` is a single thing, means "do a recalc only if there's an autoranged axis related to this attribute" - Actually, ideally we'd specify what to do if there *isn't* an autoranged axis... so like `docalcAutorange+dostyle` or something.

> Would `editType: 'calcAutorange'` make sense?

Yeah totally, good idea! And to help us out (to figure out what `editType` a new attribute should get) I should probably describe somewhere what these flags actually mean in terms of the code paths they lead to.

Reply:
Maybe `calcIfAutorange`?

> to figure out what editType a new attribute should get

FWIW, I have a history of making bad choices here for lack of really understanding the implications of some schema concepts. If you want I can try making `docs/schema.md` that explains, at the very least, what some of the schema items link up with. Some of the concepts like `_isLinkedToArray` get a little subtle. Even for the fifth time interacting with it, I think a brief description with links to a couple relevant spots in the code would help.

Reply:
> Maybe `calcIfAutorange`?

🏆 

> If you want I can try making `docs/schema.md`

Definitely time for something like that! @etpinard thoughts about how/where to do this? Should it be a standalone thing, or somehow included in `Plotly.PlotSchema.get()` or...?

Reply:
My vote would be for not including in the code so that it's as accessible and readable as possible without having to realize it's an API function and instantiate anything in order to access.

Reply:
I'd vote for adding a attribute-like edit type declaration object in [`edit_types.js`](https://github.com/plotly/plotly.js/blob/master/src/plot_api/edit_types.js) with a `description` field.  For example,

```js
var traceEditTypes: {
  docalc: {
    description: [
       // ....
    ].join(' ')
  },
  // ....
}
```

That `description` field would get stripped off by the [`compress_attributes`](https://github.com/plotly/plotly.js/blob/master/tasks/util/compress_attributes.js) transform. `Plotly.PlotSchema.get().defs.editTypes` would then return the list of edit types with their description.


I see your point @rreusser about relying an API function to show docs being less-than-ideal. But since we already have a hard time keeping https://github.com/plotly/documentation/ up to speed, adding another layer of static (e.g. a github wiki or a `docs/` directory) documentation sounds dangerous. No docs is better than bad docs, right? That said, it wouldn't hurt adding a few lines in our CONTRIBUTING.md over the edit types.